packages/manager/cypress/e2e/core/cloudpulse/widget-level-dimension-filter.spec.ts

```diff
@@ -0,0 +1,687 @@
+
+    // Define filters to add
+    const filtersWithOperators = [
+      {
+        dimension: 'entity_id',
+        operator: 'Not Equal',
+        valueToSelect: '345',
+        apiOperator: 'neq',
+        apiValue: '345',
+      },
+      {
+        dimension: 'response_type',
+        operator: 'Starts with',
+        valueToSelect: '2x',
+        apiOperator: 'startswith',
+        apiValue: '2x',
+      },
+      {
+        dimension: 'response_type',
+        operator: 'Ends with',
+        valueToSelect: '5xx',
+        apiOperator: 'endswith',
+        apiValue: '5xx',
+      },
+      {
+        dimension: 'Protocol',
+        operator: 'In',
+        valueToSelect: 'Select All',
+        apiOperator: 'in',
+        apiValue: 'TCP,UDP',
+        verifyAvailableValues: ['TCP', 'UDP'],
+      },
+      {
+        dimension: 'linode_id',
+        operator: 'Equal',
+        valueToSelect: ['Test-linode-3'],
+        apiOperator: 'eq',
+        apiValue: '789',
+      },
+    ];
+
+    // Verify global filters are excluded from dimension options (verify once)
+    ui.button.findByTitle('Add Filter').click();
+    cy.get('[data-testid="dimension_filters.0-id"]').within(() => {
+      ui.autocomplete.findByLabel('Dimension').should('be.visible').click();
+    });
+
+    globalFilters.forEach((filter) => {
+      cy.findByText(filter, { timeout: 5000 }).should('not.exist');
+    });
+
+    // Close dropdown and remove the empty filter row
+    cy.get('body').type('{esc}');
+    ui.button
+      .findByAttribute('data-testid', 'clear-icon')
+      .should('be.visible')
+      .click();
+
+    // Add filters one by one
+    filtersWithOperators.forEach(
+      (
+        { dimension, operator, valueToSelect, verifyAvailableValues },
+        index
+      ) => {
+        // Click Add Filter button
+        ui.button.findByTitle('Add Filter').click();
+
+        cy.get(`[data-testid="dimension_filters.${index}-id"]`).within(() => {
+          // Select dimension
+          ui.autocomplete
+            .findByLabel('Dimension')
+            .should('be.visible')
+            .click()
+            .type(dimension);
+          ui.autocompletePopper
+            .findByTitle(dimension)
+            .should('be.visible')
+            .click();
+
+          // Select operator
+          ui.autocomplete
+            .findByLabel('Operator')
+            .should('be.visible')
+            .type(operator);
+          ui.autocompletePopper
+            .findByTitle(operator)
+            .should('be.visible')
+            .click();
+
+          // Handle value input based on operator and value type
+          const valueSelector = `[data-qa-dimension-filter="dimension_filters.${index}-value"]`;
+
+          // Strategy 1: Handle "Select All" option
+          if (valueToSelect === 'Select All') {
+            cy.get(valueSelector)
+              .findByPlaceholderText('Select Values')
+              .should('be.visible')
+              .click();
```

Comment:
```suggestion
            cy.get(valueSelector)
              .findByPlaceholderText(valuePlaceHolder)
              .should('be.visible')
              .click();
```

then

Array.isArray(valueToSelect) ? ui.autocompletePopper
              .findByTitle(valueToSelect)
              .should('be.visible')
              .click() : cy.get(valueSelector)
              .findByPlaceholderText('Enter a Value')..type(valueToSelect as string);

Reply:
this might avoid if else branching...